### PR TITLE
Adding opinfo tests for binary cross entropy with logits

### PR DESCRIPTION
**Pull request by drisspg**
Adding the first OpInfo Request found here: #74613

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
There are a couple missing skips for TestGradient on CPU. But the structure looks good otherwise.

**Comment by drisspg** [comment]
@albanD fixed up the tests.. almost skipping more than testing lol but tried to make them as broad as possible

**Review by albanD — APPROVED** [review]
Sounds good!
Thanks!

Could you open an issue to make sure we don't forget to look into the failing tests?
